python/pyspark/worker.py

```diff
@@ -598,8 +598,18 @@ def process():
             process()
     except Exception:
         try:
+            exc_info = traceback.format_exc()
+            if sys.version_info.major < 3:
```

Comment:
Likewise, let's drop this right after we drop Python 2, which I will do right after Spark 3.